File: scrapers/sources/harashobo.pl
my $haraCount = 0;
{
	searchURL => "http://www.harashobo.com/english/selection_list.php?&page=%s&mode=0",
	searchStart => 0,
	searchEnd => 2700,
	searchIncrement => 10,

	encoding => "latin-1",

	urlExtractor => scraper {
		# process "//a[text()='>>Detail']", 'urls[]' => '@href';
		process '//img[contains(@src,"_")]', 'urls[]' => '@href';
	},

	cleanURL => sub {
		return "http://www.harashobo.com/english/selection_list.php?&page="
			. $haraCount++ . "&mode=1";
	},

	extractPageID => qr/page=(\d+)/,

	pageExtractor => scraper {
		process '//td[@class=\'txt1\']/img', source_image => sub {
				my $img = shift;
				my $src = $img->attr('src');
				$src =~ s/\.\.\///;
				$src =~ s/\d\./3\./;
				return "http://www.harashobo.com/$src";
			};
	},

	extractImageFile => qr/([^\/]+)$/
}

File: scrapers/sources/harashobo.js
exports.extract = {
	artist: "//td[@class='txt1']/br/preceding-sibling::text()[4]",
	title: "//td[@class='txt1'][3]/text()[2]", 
	date: [ '//td[@class="txt1"][3]/text()[contains(.,"1")]', function( text ) {
		return /([0-9]{4,})/.test( text ) ? RegExp.$1 : "";
	} ],
	description: '//td[@class="txt1"][3]/text()[3]',
	source_image: [ "//img[contains(@src, '_')]/@src", function( text ) {
		return "http://harashobo.com/img" + text.replace(/\.\.\/img/,'').replace(/\d\./,'3.');
	} ]
};

exports.genURL = function( id ) {
	return "http://harashobo.com/english/selection_list.php?&mode=1&page=" + id;
};
